src/io/io.cpp

```diff
@@ -1221,6 +1237,123 @@ void Write_HDF5_Field_2D_CPU(Header H, hid_t file_id, hid_t dataspace_id, float
   herr_t status = Write_HDF5_Dataset(file_id, dataspace_id, dataset_buffer, name);
 }
 
+/* \brief Before HDF5 reads data into a buffer, remap and write grid to HDF5 buffer. */
+void Fill_HDF5_Buffer_From_Grid_CPU(int nx, int ny, int nz, int nx_real, int ny_real, int nz_real, int n_ghost,
+                                    Real *hdf5_buffer, Real *grid_buffer)
+{
+  int i, j, k, id, buf_id;
+  // 3D case
+  if (nx > 1 && ny > 1 && nz > 1) {
+    for (k = 0; k < nz_real; k++) {
+      for (j = 0; j < ny_real; j++) {
+        for (i = 0; i < nx_real; i++) {
+          id                  = (i + n_ghost) + (j + n_ghost) * nx + (k + n_ghost) * nx * ny;
+          buf_id              = k + j * nz_real + i * nz_real * ny_real;
```

Comment:
Usually yes but I think in this function in particular leaving it this way makes the re-map of indices more obvious without having to look up the compute1Dindex definition (showing that HDF5 and in-memory are assuming different orderings). Since I see it both ways I'll leave it as is.

Reply:
You can just call it with the arguments in a different order. That should make the difference clear and reduces code duplication and potential for error.